website/addons/s3/static/s3NodeConfig.js

```diff
@@ -243,21 +244,66 @@ ViewModel.prototype.openCreateBucket = function() {
 
     var isValidBucket = /^(?!.*(\.\.|-\.))[^.][a-z0-9\d.-]{2,61}[^.]$/;
 
-    bootbox.prompt('Name your new bucket', function(bucketName) {
-        if (!bucketName) {
-            return;
-        } else if (isValidBucket.exec(bucketName) == null) {
-            bootbox.confirm({
-                title: 'Invalid bucket name',
-                message: 'Sorry, that\'s not a valid bucket name. Try another name?',
-                callback: function(result) {
-                    if (result) {
-                        self.openCreateBucket();
+    bootbox.dialog({
+        title: 'Create a new bucket',
+        message:
+                '<div class="row"> ' +
+                    '<div class="col-md-12"> ' +
+                        '<form class="form-horizontal"> ' +
+                            '<div class="form-group"> ' +
+                                '<label class="col-md-4 control-label" for="bucketName">Bucket Name</label> ' +
+                                '<div class="col-md-4"> ' +
+                                    '<input id="bucketName" name="bucketName" type="text" placeholder="Enter bucket\'s name" class="form-control" autofocus> ' +
+                                '</div>' +
+                            '</div>' +
+                            '<div class="form-group"> ' +
+                                '<label class="col-md-4 control-label" for="bucketLocation">Bucket Location</label> ' +
+                                '<div class="col-md-4"> ' +
+                                    '<select id="bucketLocation" name="bucketLocation" class="form-control"> ' +
+                                        '<option value="' + window.contextVars.s3Settings.defaultBucketLocationValue + '' +
+                                            '" selected>' + window.contextVars.s3Settings.defaultBucketLocationMessage + '</option> ' +
+                                        '<option value="EU">Europe Standard</option> ' +
```

Comment:
This could use refactoring. Something like:

```js
// Maps internal value used by API to display name
var BUCKET_LOCATION_MAP = {
    'EU': 'Europe Standard',
    'us-west-1': 'California',
    // ...
}

function bucketOption(value) {
   return ['<option value="', value, '">', BUCKET_LOCATION_MAP[value], '</option>'].join('');
}

//...
```